README.md

```diff
@@ -2,38 +2,68 @@
-> During the transition we retain the way the binaries were build before and adding new infrastructure to build newer versions of the binaries.
-> As such we have already added a way to build `linuxkit` and newer kernels under `linuxkit/` and `kernel/`. These new variants of the binaries are not yet
-> integrated with the top-level makefile. Please refer to their local readme files for instructions on how to build from these new variants ([linuxkit](linuxkit/README.md) and [kernel](kernel/README.md))
+The binaries being built from the sources in this package are distributed as part of [aws-nitro-enclaves-cli](https://github.com/aws/aws-nitro-enclaves-cli) and can be used to build enclave image files for AWS Nitro Enclaves.
 
-The referenced kernel in the 'linux-url' file is an example that can be used for building an enclave image. The kernel corresponding to the enclave image blobs in https://github.com/aws/aws-nitro-enclaves-cli/tree/main/blobs is based on the v4.14 Amazon Linux kernel - https://github.com/amazonlinux/linux/tree/amazon-4.14.y/master; it is different than the kernel mentioned in 'linux-url'.
+## Build
 
 ### Prerequisites
 
-The kernel download step requires setting up gpg2 with the kernel developer
-keys. Instructions are available [here](https://www.kernel.org/category/signatures.html).
+The binaries can be build through one of two ways:
+1) Build through [Nix](https://nixos.org/download.html), wich requires a local nix environment installed.
+2) Build through [Docker](https://www.docker.com/), which requires a local docker environment installed.
+
+### Building Docker Nix
+
+If you have a working nix environment installed on your machine you can build the whole set of binaries through:
 
-For Debian / Ubuntu systems, install build prequisites:
 ```
-sudo apt-get install build-essential libncurses-dev bison flex libssl-dev libelf-dev gnupg2
+nix-build -A all
 ```
 
-For Amazon Linux 2 / Fedora / RHEL / CentOS install build prequisites:
+which creates the following directory structure (Example built on x86_64):
+
 ```
-sudo yum group install "Development Tools" 
+result/
+└── x86_64
+    ├── bzImage
+    ├── bzImage.config
+    ├── init
+    ├── linuxkit
+    └── nsm.ko
 ```
 
-## Build
+Similarly you can build individual subpackages as follows:
+
+```
+nix-build -A kernel     # For `[bz]Image`, `[bz]Image.config`, `nsm.ko` (`Image` on aarch64, `bzImage` on x86_64)
+nix-build -A init       # For the bootstrap `init` binary
+nix-build -A linuxkit   # For the `linuxkit` binary
+```
+
+### Building with Docker
 
-The project can be built inside a Docker container to avoid installing toolchains and other packages
-on your local device.
+The project can be built inside a Docker container to avoid installing Nix on your local device.
+
+Build the full set of binaries through:
+
+```
+docker build -t blobs_all .
```

Comment:
`docker build -t artifacts`?

Reply:
It is just assigning a tag (or a list of tags) to the docker image for easier reference (See option `-t, --tag` for `docker build`), so we do not need to look up the image id and use that in the extract phase below.

We would need to specify `artifacts` if we were to use the `--target` option for docker build.

It is just some identifier for easier use, but I am happy to change that to any other tag that people see more fitting.

Reply:
sorry, my bad, I confused it with `--target`